pkg/e2etest/e2e_test.go

```diff
@@ -0,0 +1,529 @@
+		mockIfaces = append(mockIfaces, newPcapSource(t, fmt.Sprintf("mock%d", i+1), data))
+	}
+
+	// Run GoProbe
+	runGoProbe(t, setupSources(t, mockIfaces))
+
+	// Run GoQuery and build reference results from tracking
+	resGoQuery := runGoQuery(t, mockIfaces, 100000)
+	resReference := mockIfaces.BuildResults(t, resGoQuery)
+
+	// Counter consistency checks
+	require.Equal(t, mockIfaces.NProcessed(), int(resGoQuery.Summary.Totals.PacketsRcvd))
+	require.Equal(t, mockIfaces.NProcessed(), mockIfaces.NRead()-mockIfaces.NErr())
+
+	// Summary consistency check (do not fail yet to show details in the next check)
+	if !reflect.DeepEqual(resReference.Summary, resGoQuery.Summary) {
+		t.Errorf("Mismatch on goQuery summary, want %+v, have %+v", resReference.Summary, resGoQuery.Summary)
+	}
+
+	// Since testify creates very unreadable output when comparing the struct directly we build a stringified
+	// version of the result rows and compare that
+	refRows := make([]string, len(resReference.Rows))
+	for i := 0; i < len(refRows); i++ {
+		refRows[i] = fmt.Sprintf("%s (%s): %s %s",
+			resReference.Rows[i].Labels.Hostname, resReference.Rows[i].Labels.HostID,
+			resReference.Rows[i].Attributes,
+			resReference.Rows[i].Counters)
+	}
+	resRows := make([]string, len(resGoQuery.Rows))
+	for i := 0; i < len(resRows); i++ {
+		resRows[i] = fmt.Sprintf("%s (%s): %s %s",
+			resGoQuery.Rows[i].Labels.Hostname, resGoQuery.Rows[i].Labels.HostID,
+			resGoQuery.Rows[i].Attributes,
+			resGoQuery.Rows[i].Counters)
+	}
+	require.EqualValues(t, refRows, resRows)
+}
+
+func runGoProbe(t *testing.T, sourceInitFn func() (mockIfaces, func(c *capture.Capture) (slimcap.Source, error))) {
+
+	// We quit on encountering SIGUSR2 (instead of the ususal SIGTERM or SIGINT)
+	// to avoid killing the test
+	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGUSR2)
+	defer stop()
+
+	mockIfaces, initFn := sourceInitFn()
+	captureManager := capture.NewManager(ctx).SetSourceInitFn(initFn)
+
+	ifaceConfigs := make(config.Ifaces)
+	for _, iface := range mockIfaces {
+		ifaceConfigs[iface.name] = defaultCaptureConfig
+	}
+
+	// Wait until goProbe is done processing all packets, then kill it in the
+	// background via the SIGUSR2 signal
+	go mockIfaces.KillGoProbeOnceDone(captureManager)
+
+	captureManager.Update(ifaceConfigs, nil)
+
+	// Start goroutine for writeouts
+	writeoutHandler := writeout.NewHandler(captureManager, encoders.EncoderTypeLZ4).
+		WithPermissions(goDB.DefaultPermissions)
+
+	// Start writeout handler
+	writeoutChan := writeoutHandler.HandleWriteouts()
+
+	// Start regular rotations (minimum one second to avoid timestamp duplicates)
+	writeoutHandler.HandleRotations(ctx, time.Second)
+
+	// Wait for the interrupt signal
+	<-ctx.Done()
+
+	// Finish up
+	writeoutHandler.Close()
+	<-writeoutChan
+	captureManager.CloseAll()
+}
+
+func runGoQuery(t *testing.T, mockIfaces mockIfaces, maxEntries int) results.Result {
+
+	buf := bytes.NewBuffer(nil)
```

Comment:
For my understanding: you choose to do this vs. just capturing the part of the Query runner in order to also validate the set up boilerplate of goQuery's entrypoint, right?

If so: can't believe how far you took this. Amazing